### PR TITLE
Skip PVC expansion tests if the cluster is upgraded from OCS4.4

### LINKED ISSUE #2831: PV expansion is supported only in fresh deployments, hence, PVC resize tests should be disabled post upgrade to OCS 4.5

### PR DESCRIPTION
**Pull request by jilju**
Closes #2831 

Signed-off-by: Jilju Joy <jijoy@redhat.com>

### CODE REVIEW

**Comment by jilju** [comment]
@red-hat-storage/top-level-reviewers 
CI failed with this error.

```
INTERNALERROR> FileNotFoundError: [Errno 2] No such file or directory: 'oc': 'oc'

============================= 2 warnings in 1.32s ==============================

ERROR: InvocationError for command /home/travis/build/red-hat-storage/ocs-ci/.tox/collectonly/bin/py.test --collect-only tests (exited with code 3)
```
oc command is used to get the previous version of ocs-operator. This is causing the above failure. Is there a way to avoid this error ?

File: ocs_ci/utility/utils.py [hunk 1]
```diff
@@ -2574,3 +2574,27 @@ def check_for_rhcos_images(url):
...
+    skip_this = False
+    version_list = [version_list] if isinstance(version_list, str) else version_list
+    ocs_csv = get_ocs_csv()
+    csv_info = ocs_csv.get()
+    prev_version = csv_info.get('spec').get('replaces', '')
+    for version in version_list:
+        if f'.v{version}' in prev_version:
+            skip_this = True
+            break
+    return skip_this
```
Comment (petr-balogh) [inline ocs_ci/utility/utils.py:2600]: To be on safe side and as it's called in collection of tests and I think that there can be the case that the OCS is not installed we should wrap all this to general try: except: Exception block.  In error message write the log message if something happen and return false.

File: ocs_ci/framework/pytest_customization/marks.py [hunk 2]
```diff
@@ -192,3 +192,7 @@
 
 # Marker for skipping tests based on OCS version
 skipif_ocs_version = pytest.mark.skipif_ocs_version
+
+# Marker for skipping tests if the cluster is upgraded from a particular
+# OCS version
+skipif_upgraded_from = pytest.mark.skipif_upgraded_from
```
Comment (petr-balogh) [inline ocs_ci/framework/pytest_customization/marks.py:198]: Please register marker here:
https://github.com/red-hat-storage/ocs-ci/blob/master/pytest.ini

File: ocs_ci/utility/utils.py (hunk 1 above)
Comment (jilju) [reply ocs_ci/utility/utils.py]: Done

File: ocs_ci/framework/pytest_customization/marks.py (hunk 2 above)
Comment (jilju) [reply ocs_ci/framework/pytest_customization/marks.py:198]: Done

**Review by petr-balogh — DISMISSED** [review]

File: ocs_ci/utility/utils.py [hunk 3]
```diff
@@ -2574,3 +2574,31 @@ def check_for_rhcos_images(url):
...
+def skipif_upgraded_from(version_list):
+    """
+    This function evaluates the condition to skip a test if the cluster
+    the upgraded from a particular OCS version
+
+    Args:
+        version_list (list): List of versions to check
+
+    Return:
+        (bool): True if test needs to be skipped else False
```
Comment (vavuthu) [inline ocs_ci/utility/utils.py:2588]: leave blank line below

File: ocs_ci/utility/utils.py [hunk 4]
```diff
@@ -2574,3 +2574,31 @@ def check_for_rhcos_images(url):
     """
     r = requests.head(url)
     return r.status_code == requests.codes.ok
+
+
+def skipif_upgraded_from(version_list):
+    """
+    This function evaluates the condition to skip a test if the cluster
+    the upgraded from a particular OCS version
```
Comment (vavuthu) [inline ocs_ci/utility/utils.py:2582]:
```suggestion
    is upgraded from a particular OCS version
```

File: ocs_ci/utility/utils.py (hunk 3 above)
Comment (jilju) [reply ocs_ci/utility/utils.py:2610]: Done.

File: ocs_ci/utility/utils.py (hunk 4 above)
Comment (jilju) [reply ocs_ci/utility/utils.py]: Fixed. Thanks!

**Review by petr-balogh — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Comment by ebenahar** [comment]
@petr-balogh should we maybe merge the function in this PR with https://github.com/red-hat-storage/ocs-ci/blob/f2cec7d3c8e8df9214d3ef001d91b0ad6ce4ae86/ocs_ci/ocs/resources/ocs.py#L238?

**Review by dahorak — APPROVED** [review]

**Comment by petr-balogh** [comment]
Jilju verified locally:  https://paste.centos.org/view/318063be

**Comment by jilju** [comment]
skipif_upgraded_from marker is verified locally with ```@skipif_upgraded_from(['4.5'])``` in tests.

[verify_skipif_upgraded_from.txt](https://github.com/red-hat-storage/ocs-ci/files/5174365/verify_skipif_upgraded_from.txt)

**Comment by petr-balogh** [comment]
> @petr-balogh should we maybe merge the function in this PR with
> 
> https://github.com/red-hat-storage/ocs-ci/blob/f2cec7d3c8e8df9214d3ef001d91b0ad6ce4ae86/ocs_ci/ocs/resources/ocs.py#L238
> 
> ?

@jilju can you maybe take this in new PR once you have time? You can open the issue about it to not forget. Thanks

**Comment by jilju** [comment]
> > @petr-balogh should we maybe merge the function in this PR with
> > https://github.com/red-hat-storage/ocs-ci/blob/f2cec7d3c8e8df9214d3ef001d91b0ad6ce4ae86/ocs_ci/ocs/resources/ocs.py#L238
> > 
> > ?
> 
> @jilju can you maybe take this in new PR once you have time? You can open the issue about it to not forget. Thanks

@petr-balogh 
Opened https://github.com/red-hat-storage/ocs-ci/issues/2872

**Pull request merged**